### PR TITLE
Show all BP passports in patient edit screen

### PR DESCRIPTION
**Pull request by rakshakhegde**
https://www.pivotaltracker.com/story/show/171292061

This PR is only showing the list of all BP passports in patient edit screen without the add BP passport button.

### CODE REVIEW

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt [hunk 1]
```diff
@@ -104,6 +113,17 @@ class EditPatientEffectHandler @AssistedInject constructor(
     }
   }
 
+  private fun fetchBpPassports(scheduler: Scheduler) =
+      ObservableTransformer<FetchBpPassportsEffect, EditPatientEvent> { fetchBpPassportsEffect ->
+        fetchBpPassportsEffect.subscribeOn(scheduler)
+            .flatMap { patientRepository.patientProfile(it.patientUuid) }
+            .filterAndUnwrapJust()
+            .map { patientProfile ->
+              patientProfile.withoutDeletedBusinessIds().businessIds.filter { it.identifier.type == BpPassport }
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt:122]: You can add an `subscribeOn` here as well.

Comment (sanchita) [reply app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt:122]: Please reformat to 
```
    patientProfile
        .withoutDeletedBusinessIds()
        .businessIds
        .filter { it.identifier.type == BpPassport }
```

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientEvent.kt [hunk 2]
```diff
@@ -1,10 +1,13 @@
 package org.simple.clinic.editpatient
 
 import org.simple.clinic.patient.Gender
+import org.simple.clinic.patient.businessid.BusinessId
 import org.simple.clinic.widgets.UiEvent
 
 sealed class EditPatientEvent : UiEvent
 
+data class BpPassportsFetchedEvent(val bpPasssports: List<BusinessId>) : EditPatientEvent()
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientEvent.kt:9]: We don't need to suffix `Event` here.

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt [hunk 3]
```diff
@@ -56,16 +58,23 @@ class EditPatientEffectHandler @AssistedInject constructor(
     return RxMobius
         .subtypeEffectHandler<EditPatientEffect, EditPatientEvent>()
         .addConsumer(PrefillFormEffect::class.java, ::prefillFormFields, schedulersProvider.ui())
+        .addConsumer(DisplayBpPassportsEffect::class.java, { displayBpPassports(it.bpPassports) }, schedulersProvider.ui())
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt:61]: Please add a test for this effect in `EditPatientEffectHandlerTest`.

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt [hunk 4]
```diff
@@ -56,16 +58,23 @@ class EditPatientEffectHandler @AssistedInject constructor(
...
         .addConsumer(ShowValidationErrorsEffect::class.java, ::showValidationErrors, schedulersProvider.ui())
         .addConsumer(HideValidationErrorsEffect::class.java, { ui.hideValidationErrors(it.validationErrors) }, schedulersProvider.ui())
         .addAction(ShowDatePatternInDateOfBirthLabelEffect::class.java, ui::showDatePatternInDateOfBirthLabel, schedulersProvider.ui())
         .addAction(HideDatePatternInDateOfBirthLabelEffect::class.java, ui::hideDatePatternInDateOfBirthLabel, schedulersProvider.ui())
         .addAction(GoBackEffect::class.java, ui::goBack, schedulersProvider.ui())
         .addAction(ShowDiscardChangesAlertEffect::class.java, ui::showDiscardChangesAlert, schedulersProvider.ui())
+        .addTransformer(FetchBpPassportsEffect::class.java, fetchBpPassports(schedulersProvider.io()))
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt:68]: Please add a test for this effect in EditPatientEffectHandlerTest.

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt [hunk 5]
```diff
@@ -295,6 +297,19 @@ class EditPatientScreen(context: Context, attributeSet: AttributeSet) : Relative
 
   private fun ageTextChanges(): Observable<EditPatientEvent> = ageEditText.textChanges(::AgeChanged)
 
+  override fun displayBpPassports(identifiers: List<String>) {
+    bpPassportsContainer.removeAllViews()
+    identifiers.forEach { identifier ->
+      val bpPassportView = LayoutInflater.from(context).inflate(R.layout.patient_edit_bp_passport_view, bpPassportsContainer, false)
+      bpPassportView.bpPassportIdentifier.text = identifier
+      bpPassportsContainer.addView(bpPassportView)
```
Comment (sanchita) [inline app/src/main/java/org/simple/clinic/editpatient/EditPatientScreen.kt:305]: Can we extract the inflation of the view to a separate method?

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt (hunk 1 above)
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt:122]: Done

File: app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt [hunk 6]
```diff
@@ -104,6 +113,17 @@ class EditPatientEffectHandler @AssistedInject constructor(
     }
   }
 
+  private fun fetchBpPassports(scheduler: Scheduler) =
+      ObservableTransformer<FetchBpPassportsEffect, EditPatientEvent> { fetchBpPassportsEffect ->
+        fetchBpPassportsEffect.subscribeOn(scheduler)
```
[the remark this answers is not in the document]
Comment (rakshakhegde) [reply app/src/main/java/org/simple/clinic/editpatient/EditPatientEffectHandler.kt:118]: Done

**Pull request merged**